kubeflow/openmpi/assets.libsonnet

```diff
@@ -0,0 +1,128 @@
+{
+  all(params):: {
+    kind: "ConfigMap",
+    apiVersion: "v1",
+    metadata: {
+      name: "openmpi-assets",
+      labels: {
+        app: params.name,
+      },
+    },
+    data: {
+      "gen_hostfile.sh": |||
+        set -exv
+
+        # Download kubectl
+        apk add --update ca-certificates
+        apk add --update -t deps curl
+        curl -L https://storage.googleapis.com/kubernetes-release/release/v1.9.6/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl
```

Comment:
Why are you installing kubectl in an init script rather than creating a Docker image?
I think baking into the DockerImage will be more performant and more reliable than downloading it via the init container.

Reply:
I wanted to make the app generic so that we can deploy any docker image (with openmpi installed) using this app. For example, it seems strange to me to modify the horovod package to install kubectl. With this approach, I can deploy the horovod image with no change.

Reply:
It looks like the sole purpose of this script is to generate the host file. If that's the case is there any reason to run it in the same container as the workers? You could run it in a separate pod and create a configmap that would then be mounted into the workers (you could also do it in an init container).

This way you wouldn't have to modify the actual worker image and you could have a separate docker image that could have the code for generating the host file.